File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        results = []
        for rows in range(numRows):
            # size = rows + 1
            # begining and end are 1
            row = [1] * (rows + 1)
            for cols in range(1, rows):
                # dp[i][j] = dp[i-1][j-1] + dp[i-1][j]
                row[cols] = results[rows-1][cols-1] + results[rows-1][cols]
            results.append(row)
            
        return results

File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> results;
        results.push_back({1});
        if (numRows == 1) return results;
        results.push_back({1, 1});
        if (numRows == 2) return results;
        for (int row=2; row<numRows; row++) {
            results.push_back({1});
            for (int i=0; i<results[row-1].size()-1; i++) {
                results[row].push_back( results[row-1][i] + results[row-1][i+1] );
            }
            results[row].push_back(1);
        }
        return results;
    }
};